java/vector/src/main/java/org/apache/arrow/vector/types/pojo/Field.java

```diff
@@ -111,9 +112,26 @@ public FieldVector createVector(BufferAllocator allocator) {
    * Constructs a new instance from a flatbuffer representation of the field.
    */
   public static Field convertField(org.apache.arrow.flatbuf.Field field) {
+    Map<String, String> metadata = new HashMap<>();
+    for (int i = 0; i < field.customMetadataLength(); i++) {
+      KeyValue kv = field.customMetadata(i);
+      String key = kv.key();
+      String value = kv.value();
+      metadata.put(key == null ? "" : key, value == null ? "" : value);
```

Comment:
I think we might shade guava in which case you could use: https://google.github.io/guava/releases/23.0/api/docs/com/google/common/base/Strings.html#nullToEmpty-java.lang.String-

Reply:
We don't seem to include it inside arrow-vector, do we want to do that? (Don't think it's worth it for one method...)

Reply:
No, I we might have just copied the methods we needed instead of shading it.